src/nupic/frameworks/opf/common_models/cluster_params.py

```diff
@@ -0,0 +1,139 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import csv
+import json
+import numpy as np
+import os
+from pkg_resources import resource_stream
+
+
+def setRandomEncoderParams(params, minVal, maxVal, minResolution):
+  """
+  Given model params, figure out the correct parameters for the
+  RandomDistributed encoder. Modifies params in place.
+  """
+  encodersDict= (
+    params['modelConfig']['modelParams']['sensorParams']['encoders']
+    )
+  for v in encodersDict.itervalues():
+    if v is not None:
+      if v['type'] == 'RandomDistributedScalarEncoder':
+        resolution = max(minResolution,
+                         (maxVal - minVal) / v.pop('numBuckets')
+                        )
+        encodersDict['c1']['resolution'] = resolution
+
+
+
+def getScalarMetricWithTimeOfDayParams(metricData,
+                                       minVal=None,
+                                       maxVal=None,
+                                       minResolution=None,
+                                       useRandomEncoder=True):
+  """
+    Return an ordered list of JSON strings that can be used to create an
+    anomaly model for OPF's ModelFactory.
+
+
+    Args:
+        metricData:  numpy array of metric data. Used to calculate minVal
+                     and maxVal if either is unspecified
+
+        minVal:      minimum value of metric. Used to set up encoders. If None
+                     will be derived from metricData.
+
+        maxVal:      maximum value of metric. Used to set up input encoders. If
+                     None will be derived from metricData
+
+        minResolution: minimum resolution of metric. Used to set up encoders.
+                       If None, will use default value of 0.001.
+
+        useRandomEncoder: if True, use RandomDistributedScalarEncoder instead
+                     of ScalarEncoder
+
+    Assumptions:
+        The timeStamp field corresponds to c0
+        The predicted field corresponds to c1
+  """
+  # Default values
+  if minResolution is None:
+    minResolution = 0.001
+
+  # Read a sorted list of parameters from the appropriate directory
+  if useRandomEncoder:
+    paramsDirectory = 'anomaly_params_random_encoder'
+  else:
+    # This will go away soon MER-1442
+    paramsDirectory = 'scalarMetricWithTimeOfDayAnomalyParams'
+
+  with resource_stream('nupic.frameworks.opf.common_models',
+                       os.path.join(paramsDirectory,
+                                    'paramOrder.csv')) as fileObj:
```

Comment:
@rhyolight, that thought came to my mind as I was driving home last night, and I made a mental note to check on it this morning. This didn't get caught, since I tested with nupic installed via `setup.py develop`.